.buildbot.sh

```diff
@@ -2,12 +2,25 @@
 #
 # Build script for continuous integration.
 
-./x.py clean  # We don't clone afresh to save time and bandwidth.
-git clean -dffx # If upstream removes a submodule, remove the files from disk.
+set -e
 
 # Ensure the build fails if it uses excessive amounts of memory.
 ulimit -d $((1024 * 1024 * 8)) # 8 GiB
 
 # Note that the gdb must be Python enabled.
 /usr/bin/time -v env PATH=/opt/gdb-8.2/bin:${PATH} \
     RUST_BACKTRACE=1 ./x.py test --config .buildbot.toml
+
+# Archive the build and put it in /opt
+GIT_HASH=`git rev-parse --short HEAD`
+TARBALL_TOPDIR=ykrustc-stage2
+TARBALL_NAME=ykrustc-stage2-${GIT_HASH}.tar.bz2
+SNAP_DIR=/opt/ykrustc-bin-snapshots
```

Comment:
I never know what `opt` is supposed to mean on Linux. Is it a directory the OS / distribution reserves for itself? In other words, can we guarantee we're not causing some weird future problem?

Reply:
`/opt` is a place for stuff you installed manually. We already rely on this a lot for various buildbot software and dependencies, for example buildbot itself is intalled in a virtualenv under /opt.

Strictly speaking the snapshot dir isn't super-suited to `/opt` as the binary tarballs are not immediately executable from that location. Feel free to propose an alternative, but I'm inclined to keep all of the buildbot dependencies in one location. I already have a README file in `/opt` describing the meaning of each sub-dir.

https://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/opt.html

Reply:
(BTW /opt is nothing to do with Linux. I first learned of this convention in Solaris 8)

Reply:
Some OSs / distributions install packages under `/opt`. So long as our machines don't do that, I'm fine with using that, at least in the short term.

Reply:
I've never seen `/opt` used by a package manager. I think we are OK on Debian for sure.

Reply:
OK.